src/optimal_retention.rs

```diff
@@ -682,6 +708,7 @@ fn sample<F>(
     n: usize,
     progress: &mut F,
     cards: &Option<Vec<Card>>,
+    target: &CMRRTargetFunction,
```

Comment:
```suggestion
    CMRRTargetFunction(target): &CMRRTargetFunction,
```

I guess we can try like this.